cns/restserver/api.go

```diff
@@ -1415,3 +1415,45 @@ func (service *HTTPRestService) deleteHostNCApipaEndpoint(w http.ResponseWriter,
 	err = service.Listener.Encode(w, &response)
 	logger.Response(service.Name, response, response.Response.ReturnCode, ReturnCodeToString(response.Response.ReturnCode), err)
 }
+
+// This function is used to query NMagents's supproted APIs list
+func (service *HTTPRestService) nmAgentSupportedApisHandler(w http.ResponseWriter, r *http.Request) {
+	logger.Printf("[Azure CNS] nmAgentSupportedApisHandler")
+	logger.Request(service.Name, "nmAgentSupportedApisHandler", nil)
+	var (
+		err           error
+		req           cns.NmAgentSupportedApisRequest
+		returnCode    int
+		returnMessage string
+		supportedApis []string
+	)
+
+	err = service.Listener.Decode(w, r, &req)
+	logger.Request(service.Name, &req, err)
+	if err != nil {
+		return
+	}
+
+	switch r.Method {
+	case "POST":
+		supportedApis, returnMessage = nmagentclient.GetNmAgentSupportedApis(
+			req.GetNmAgentSupportedApisURL)
+		if returnMessage != "" {
+			returnCode = NmAgentSupportedApisError
+			logger.Errorf("[Azure-CNS] %s", returnMessage)
+		}
+
+	default:
+		returnMessage = "[Azure-CNS] GetHostLocalIP API expects a GET."
```

Comment:
doesn't this API expect a `POST`?

Reply:
Right, I modeled it as GET first, changed it to POST but forgot to update this message.